src/preferences/configobject.cpp

```diff
@@ -186,34 +187,66 @@ template <class ValueType> void ConfigObject<ValueType>::reopen(const QString& f
     }
 }
 
-template <class ValueType> void ConfigObject<ValueType>::save() {
+/// Save the ConfigObject to disk.
+/// Returns true on success
+template<class ValueType>
+bool ConfigObject<ValueType>::save() {
     QReadLocker lock(&m_valuesLock); // we only read the m_values here.
-    QFile file(m_filename);
-    if (!QDir(QFileInfo(file).absolutePath()).exists()) {
-        QDir().mkpath(QFileInfo(file).absolutePath());
+    QFile tmpFile(m_filename + kTempFilenameExtension);
+    if (!QDir(QFileInfo(tmpFile).absolutePath()).exists()) {
+        QDir().mkpath(QFileInfo(tmpFile).absolutePath());
+    }
+    if (!tmpFile.open(QIODevice::WriteOnly | QIODevice::Text)) {
+        qWarning() << "Could not write config file: " << tmpFile.fileName();
+        return false;
+    }
+    QTextStream stream(&tmpFile);
+    stream.setCodec("UTF-8");
+
+    QString group = "";
+
+    uint minLength = 0;
+    for (auto i = m_values.constBegin(); i != m_values.constEnd(); ++i) {
+        //qDebug() << "group:" << it.key().group << "item" << it.key().item << "val" << it.value()->value;
+        if (i.key().group != group) {
+            group = i.key().group;
+            stream << "\n"
+                   << group << "\n";
+            minLength += i.key().group.length() + 2;
+        }
+        stream << i.key().item << " " << i.value().value << "\n";
+        minLength += i.key().item.length() + i.value().value.length() + 1;
     }
-    if (!file.open(QIODevice::WriteOnly | QIODevice::Text)) {
-        qDebug() << "Could not write file" << m_filename << ", don't worry.";
-        return;
-    } else {
-        QTextStream stream(&file);
-        stream.setCodec("UTF-8");
 
-        QString grp = "";
+    stream.flush();
+    tmpFile.flush();
```

Comment:
Duplicate flush

> If you call flush(), QTextStream will empty all data from its write buffer into the device and call flush() on the device.

Reply:
same here, this isn't resolved yet.

Reply:
@poelzi I am not going to do any future reviews if comments are ignored and I have to double check everything again. Not possible.

Reply:
Maybe he just forgot to push?

Reply:
Yes, probably ;) I must admit that I marked issues as resolved before pushing.

Reply:
Same.

Reply:
yes :(